webapp/packages/plugin-connections-administration/src/ConnectionForm/ConnectionAccess/ConnectionFormAccessPart.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { FormPart, formStatusContext, type IFormState } from '@cloudbeaver/core-ui';
+import type { IExecutionContextProvider } from '@cloudbeaver/core-executor';
+import type { IConnectionFormAccessState } from './IConnectionFormAccessState.js';
+import { createConnectionParam, type ConnectionInfoResource } from '@cloudbeaver/core-connections';
+import { action, makeObservable } from 'mobx';
+import type { IConnectionFormState } from '@cloudbeaver/plugin-connections';
+import { getSubjectDifferences } from '@cloudbeaver/core-utils';
+
+const getDefaultState = () =>
+  ({
+    grantedSubjects: [],
+    editing: false,
+  }) as IConnectionFormAccessState;
+
+export class ConnectionFormAccessPart extends FormPart<IConnectionFormAccessState, IConnectionFormState> {
+  constructor(
+    formState: IFormState<IConnectionFormState>,
+    private readonly connectionInfoResource: ConnectionInfoResource,
+  ) {
+    super(formState, getDefaultState());
+
+    makeObservable(this, {
+      edit: action.bound,
+      revoke: action.bound,
+      grant: action.bound,
+    });
+  }
+
+  protected override async loader(): Promise<void> {
+    const connectionId = this.formState.state.config.connectionId;
+    const projectId = this.formState.state.projectId;
+
+    if (!connectionId || !projectId) {
+      this.setInitialState(getDefaultState());
+      return;
+    }
+
+    const key = createConnectionParam(projectId, connectionId);
+    const subjects = await this.connectionInfoResource.loadAccessSubjects(key);
+
+    this.setInitialState({
+      ...getDefaultState(),
+      grantedSubjects: subjects.map(subject => subject.subjectId),
+    });
+  }
+
+  protected override async saveChanges(
+    data: IFormState<IConnectionFormState>,
+    contexts: IExecutionContextProvider<IFormState<IConnectionFormState>>,
+  ): Promise<void> {
+    const status = contexts.getContext(formStatusContext);
+    const connectionId = this.formState.state.config.connectionId;
+
+    if (this.formState.state.submitType === 'test' || !data.state.projectId || !status.saved || !connectionId || !this.loaded) {
+      return;
+    }
+
+    const key = createConnectionParam(data.state.projectId, connectionId);
+
+    const currentGrantedSubjects = await this.connectionInfoResource.loadAccessSubjects(key);
+    const currentGrantedSubjectIds = currentGrantedSubjects.map(subject => subject.subjectId);
```

Comment:
You don't need to call this function because the initial state will be updated by loader fn before saving (the form part calls it)
so use `this.initialState.grantedSubjects `